src/solver/variable/setofareas.hxx

```diff
@@ -227,7 +227,9 @@ inline void SetsOfAreas<NextT>::buildSurveyReport(SurveyResults& results,
                                                   int fileLevel,
                                                   int precision) const
 {
-    if (count && (dataLevel & Category::setOfAreas))
+    int count_int = count;
+    bool setOfAreasDataLevel = dataLevel & Category::setOfAreas;
+    if (count_int && setOfAreasDataLevel)
```

Comment:
Suggestion :

```cpp
if ((int)count && (dataLevel & Category::setOfAreas))
```

Reply:
As I can remember, I tried it, but it did not make the warning go away.

Reply:
what is the warning ?